### DISCUSSION TITLE
Mismatching worker version with pdfjs-dist as a direct dependency (answered ✅)

### DISCUSSION THREAD

**Discussion by lirbank** [wojtekmaj/react-pdf#1520] [Q&A] [answered]
If you've installed `pdfjs-dist` as a direct dependency and the version is not matching the `pdfjs-dist` that `react-pdf` depends on, you may get an error that looks like this:

```
Warning: UnknownErrorException: The API version "3.6.172" does not match the Worker version "3.7.107".
```

In this case I have this `"pdfjs-dist": "3.7.107"` in my package.json, but `react-pdf` depends on `"pdfjs-dist": "3.6.172"`. To solve this, make sure to use the worker provided by the `pdfjs-dist` version that `react-pdf` specifically depends on, eg:

```ts
pdfjs.GlobalWorkerOptions.workerSrc = new URL(
  // "pdfjs-dist/build/pdf.worker.min.js",
  "react-pdf/node_modules/pdfjs-dist/build/pdf.worker.min.js",
  import.meta.url
).toString();
```

Unfortunately you have to change this back to "pdfjs-dist/build/pdf.worker.min.js" if the direct dependency version matches react-pdf's version, since NPM will move `pdfjs-dist` to the top level node_modules if the versions match (eg if I were to change my package.json to `"pdfjs-dist": "3.6.172"` in the example above).

I am using Next.js 13 and NPM 9.

**Comment by wojtekmaj** [comment]
Thanks for the tip!

My recommendation would be to also keep these versions in sync and avoid duplication - pdfjs-dist is suuuuuper heavy package, so it matters more than usual.

**Reply by lirbank** [reply]
Good point! Do you think react-pdf's dependency on pdfjs-dist be relaxed a bit? Right now it's an exact dependency so one could not upgrade pdfjs-dist to the latest version before react-pdf is upgraded.

**Reply by lirbank** [reply]
Eg. changing this 

https://github.com/wojtekmaj/react-pdf/blob/main/package.json#LL42C8-L42C8

from

```
  "pdfjs-dist": "3.6.172",
```
to
```
  "pdfjs-dist": ">=3.6.172",
```
or something like that.

**Reply by wojtekmaj** [reply, marked as answer]
I would love to, but historically this lead to significant issues. PDF.js is released with complete disregard for semver. This means that ANY PDF.js version update that has not been tested can potentially render React-PDF useless. Or throwing gazillions of warnings at best.

I think that's the only exception to using `^` range I've made across all my 50+ packages.
